agent/engine/docker_task_engine.go

```diff
@@ -1266,10 +1268,16 @@ func (engine *DockerTaskEngine) startContainer(task *apitask.Task, container *ap
 	}
 	if task.IsExecCommandAgentEnabled() {
 		if err := engine.execCmdMgr.StartAgent(engine.ctx, engine.client, task, container, dockerID); err != nil {
-			seelog.Errorf("Task engine [%s]: Failed to start ExecCommandAgent Process for container [%s]: %v", task.Arn, dockerID, err)
-			// TODO: [ecs-exec] call control plane to report ExecCommandAgent FAIL here
+			seelog.Errorf("Task engine [%s]: Failed to start ExecCommandAgent Process for container [%s]: %v", task.Arn, container.Name, err)
+		}
+		// whether we started or failed to start, we'll want to emit a state change event
+		// redundant state change events like RUNNING->RUNNING are allowed
+		mTask, ok := engine.managedTasks[task.Arn]
+		if ok {
+			mTask.emitContainerEvent(mTask.Task, container, "")
+		} else {
+			seelog.Errorf("Task engine [%s]: Failed to update status of ExecCommandAgent Process for container [%s]: %v", task.Arn, container.Name, err)
```

Comment:
I think there will be no `err` here. Maybe this should say just `seelog.Errorf("Task engine [%s]: Failed to update status of ExecCommandAgent Process for container [%s]: managed task not found", task.Arn, container.Name)`

Reply:
I'll change this with my next PR to add in the updated model.